### PR TITLE
Add aten::topk and its variants

### PR DESCRIPTION
**Pull request by xytintel**
Task list:
- [x] topk
- [x] topk.values

### CODE REVIEW

File: src/ATen/native/xpu/TensorTopK.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    int64_t k,
+    int64_t dim_,
+    bool largest,
+    bool sorted,
+    Tensor& values,
+    Tensor& indices) {
+  int64_t dim = maybe_wrap_dim(dim_, self.dim(), /*wrap_scalar=*/true);
+  TORCH_CHECK(
+      k >= 0 && k <= (self.dim() > 0 ? self.size(dim) : 1),
+      "selected index k out of range");
```
Comment (fengyuan14) [inline src/ATen/native/xpu/TensorTopK.cpp:57]: Do we support `k == 0`? CUDA returns empty tensor directly?
image

File: src/ATen/native/xpu/sycl/SortingKernels.h [hunk 2]
```diff
@@ -403,6 +403,124 @@ void segmented_radix_sort_pairs_kernel(
...
+        k_,
+        keys_out_ + seg_idx * k_,
+        values_out_ + seg_idx * k_);
+  }
+
+  void sycl_ker_config_convention(sycl::handler& cgh) {
+    slm_ = sycl_local_acc_t<char>(
+        make_alignment_n<MAX_KV_BYTES>(method_t::LocalMemorySize()) +
+            make_alignment_n<MAX_KV_BYTES>(k_ * sizeof(key_t)) +
+            k_ * sizeof(value_t),
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/SortingKernels.h:464]: Do we need to check the size, which should be less than hardware limitation?

Comment (xytintel) [reply src/ATen/native/xpu/sycl/SortingKernels.h:464]: No need, LocalMemorySize returns a fixed small value, k_ is limited within the work group size.

File: src/ATen/native/xpu/TensorTopK.cpp (hunk 1 above)
Comment (xytintel) [reply src/ATen/native/xpu/TensorTopK.cpp:57]: Yes, we need

Comment (fengyuan14) [reply src/ATen/native/xpu/TensorTopK.cpp:57]: Any change?

**Comment by fengyuan14** [comment]
All topk cases pass. Please retrigger the pre-ci after addressing comments.
image

**Review by fengyuan14 — APPROVED** [review]

File: src/ATen/native/xpu/TensorTopK.cpp (hunk 1 above)
Comment (xytintel) [reply src/ATen/native/xpu/TensorTopK.cpp:57]: Already fixed

**Pull request merged**